presto-kudu/src/main/java/io/prestosql/plugin/kudu/TypeHelper.java

```diff
@@ -156,7 +158,9 @@ public static Object getJavaValue(Type type, Object nativeValue)
             return ((Slice) nativeValue).toByteBuffer();
         }
         if (type instanceof DecimalType) {
-            return nativeValue;
+            DecimalType dt = (DecimalType) type;
+            return new BigDecimal(BigInteger.valueOf((Long) nativeValue),
```

Comment:
`(Long) nativeValue` will throw ClassCastException if the value is larger than max value of long. We can avoid the situation by;
```java
            DecimalType decimalType = (DecimalType) type;
            if (decimalType.isShort()) {
                return new BigDecimal(BigInteger.valueOf((long) nativeValue), decimalType.getScale());
            }
            return new BigDecimal(decodeUnscaledValue((Slice) nativeValue), decimalType.getScale());
```

tips: We don't use abbreviations (`dt`)

Reply:
Thank you for your help, I really appreciate that! [handshake]